### PR TITLE
Replicate Environment Variable Resolution/Prompting like the current integration with the Extension

### PR DESCRIPTION
**Pull request by pastuxso**
TODO: Desecription

Closes: #543

### CODE REVIEW

File: internal/cmd/run.go [hunk 1]
```diff
@@ -203,6 +199,8 @@ func runCmd() *cobra.Command {
 					return err
 				}
 
+				defer localRunner.Stop()
```
Comment (pastuxso) [inline internal/cmd/run.go:202]: It's required to stop the Ad-Hoc gRPC server

File: internal/cmd/common.go [hunk 2]
```diff
@@ -350,63 +350,56 @@ func promptEnvVars(cmd *cobra.Command, envs []string, tasks ...project.Task) err
 	for _, task := range tasks {
 		block := task.CodeBlock
 
-		if block.PromptEnv() {
-			varPrompts := getCommandExportExtractMatches(block.Lines())
-			for _, ev := range varPrompts {
-				if slices.Contains(keys, ev.Key) {
-					block.SetLine(ev.LineNumber, "")
+		if !block.PromptEnv() {
```
Comment (pastuxso) [inline internal/cmd/common.go:353]: Refactor in favor of an early return instead of a nested block

File: internal/runner/client/client_local.go [hunk 3]
```diff
@@ -218,6 +171,43 @@ func shellID() (int, bool) {
 	return i, true
 }
 
+func (r *LocalRunner) setupAdHocServer() (
```
Comment (sourishkrout) [inline internal/runner/client/client_local.go:174]: We want to extend the existing [Runner interface](https://github.com/stateful/runme/blob/main/internal/runner/client/client.go#L55-L66) that separates Local and Remote implementations, not make the Local implementation Remote.

**Review by sourishkrout — CHANGES_REQUESTED** [review]
See in-code comment. We also don't want to redo session mgmt for the clients either.

**Pull request closed without merging**